### PR TITLE
Bugfix/unique test search term report

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**
https://github.com/fivetran/dbt_apple_search_ads/issues/16

**This PR will result in the following new package version:**

v0.2.2

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**
Adds new fields to the unique test for the `apple_search_ads__search_term_report` model:

            - keyword_id
            - ad_group_id
            - campaign_id
            - organization_id
            - match_type

In addition to the existing `search_term_text` and `date_day`

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [ n/a] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

dbt test passes. Again our internal data used for development is light and therefore didn't catch this issue before, but the customer who contributed this PR has confirmed that the fields they added now allows the unique test to pass.

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [ n/a] README updates have been applied (if applicable)
    
- [n/a ] DECISIONLOG updates have been updated (if applicable)
- [n/a ] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

**Comment by fivetran-reneeli** [comment]
Thank you @yuna-tang for contributing this PR! We will be reviewing this and including this update to the package. 

Additional changes:
- upgraded integration test version
- updated CHANGELOG
- Also added `match_type` and `organization_id` to the unique test to mirror what we did in the [ad reporting rollup package](https://github.com/fivetran/dbt_ad_reporting/blob/cb0738c6c5b9b7bf2c8598638816d3401ae06fc0/models/ad_reporting_models.yml#L183-L195).

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+# dbt_apple_search_ads v0.2.1
+## Bugfix:
+- Added the following additional columns in dbt_utils.unique_combination_of_columns test for `apple_search_ads__search_term_report`  ([PR #18](https://github.com/fivetran/dbt_apple_search_ads/pull/18)):
 
- ## Under the Hood:
+        - keyword_id
+        - ad_group_id
+        - campaign_id
+        - organization_id
+        - match_type
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:9]: Can we update this changelog entry to include all the fields that are being included in the test. Maybe something long the lines of

```md
- Updated the dbt_utils.unique_combination_of_columns test for the `apple_search_ads__search_term_report` to include the following fields. ([PR #18](https://github.com/fivetran/dbt_apple_search_ads/pull/18)):
        - date_day
        - search_term_text
        - keyword_id
        - ad_group_id
        - campaign_id
        - organization_id
        - match_type
```

File: models/apple_search_ads.yml [hunk 2]
```diff
@@ -235,7 +235,12 @@ models:
       - dbt_utils.unique_combination_of_columns:
           combination_of_columns:
             - search_term_text
+            - match_type
             - date_day
+            - keyword_id
+            - ad_group_id
+            - campaign_id
+            - organization_id
```
Comment (fivetran-joemarkiewicz) [inline models/apple_search_ads.yml:243]: I noticed you added the `organization_id` and `match_type` fields to the uniqueness test, which the customer didn't have in their initial PR. My instinct would be to include only the fields we are marking as keys from the ERD of Apple Search Ads. Was there additional information that informed you to add the above two fields?

image

Comment (fivetran-joemarkiewicz) [reply models/apple_search_ads.yml:243]: If the above is updated we should also remember to update the CHANGELOG.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli thanks for working through this PR and incorporating the customers changes from the accompanying PR!

I do have a few comments in the review that I would like your eyes on to respond or make updates. Additionally, a few other comments I had can be found below:
- It looks like the docs have not been generated. After making any necessary updates (especially if one is made for the fields to test), be sure to regen the docs.
- Small note on the PR submission form. If a checklist item does not pertain to the PR you are working on, can you update the checkbox to be `[n/a]`. This way it does not show as an empty checkbox.

File: dbt_apple_search_ads
[the remark this answers is not in the document]
Comment (fivetran-reneeli) [reply dbt_apple_search_ads:1]: Not sure how that got there. Thanks for catching, removed

File: models/apple_search_ads.yml (hunk 2 above)
Comment (fivetran-reneeli) [reply models/apple_search_ads.yml:243]: Yes `organization_id` and `match_type` were added to keep standard to what we use to test in the [ad reporting package](https://github.com/fivetran/dbt_ad_reporting/blob/cb0738c6c5b9b7bf2c8598638816d3401ae06fc0/models/ad_reporting_models.yml#L183-L195). I think it makes sense to go based off what's in the ERD (pending us being notified what fivetran_id is hashed from) though. 

But do you think there's any downside to adding those fields as well?

**Comment by fivetran-reneeli** [comment]
Updated PR and will wait to regen docs when everything is finalized.

@fivetran-joemarkiewicz ready for re-review

File: models/apple_search_ads.yml (hunk 2 above)
Comment (fivetran-joemarkiewicz) [reply models/apple_search_ads.yml:243]: I don't foresee there being any downside with including them. Especially if we already test on these fields uniqueness in the downstream ad reporting model. I am comfortable keeping the test with how you have it in this PR.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-reneeli these changes look good to go! Just be sure to regenerate the docs before kicking off the release review process.

**Comment by yuna-tang** [comment]
@fivetran-reneeli , I noticed the release version for dbt_apple_search_ads will be v0.2.1. Do you know which version of `fivetran/ad_reporting` will include this change?

**Comment by fivetran-reneeli** [comment]
Hi @yuna-tang , it should automatically pull this latest version given the [parameters](https://github.com/fivetran/dbt_ad_reporting/blob/cb0738c6c5b9b7bf2c8598638816d3401ae06fc0/packages.yml#L2-L3)  currently are [">=0.2.0", "<0.3.0"]!

Note that we moved this to v0.2.2 because I made an accidental release on v0.2.1, but same effect should apply! This has been merged to main so will be closing this out

**Pull request merged**